types/mapper.go

```diff
@@ -0,0 +1,232 @@
+package types
+
+import (
+	"reflect"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	crypto "github.com/tendermint/go-crypto"
+	amino "github.com/tendermint/go-amino"
+)
+
+// Manages utxo's in existence
+// Uses go-amino encoding/decoding library
+// Does not need to be changed to RLP
+type UTXOMapper struct {
+	
+	// The key used to access the store from the Context.
+	key sdk.StoreKey
+
+	// The prototypical UTXO concrete type
+	proto UTXOHolder
+
+	// The Amino codec for binary encoding/decoding of utxo's 
+	cdc *amino.Codec 
+}
+
+// NewUTXOMapper returns a new UtxoMapper that
+// uses go-Amino to (binary) encode and decode concrete UTXO
+func NewUTXOMapper(key sdk.StoreKey, proto UTXOHolder) UTXOMapper {
+	cdc := amino.NewCodec()
+	Register(cdc)
+	return UTXOMapper {
+		key:   key,
+		proto: proto,
+		cdc:   cdc, 
+	}
+
+}
+
+// Create and return a sealed utxo mapper. Not sure if necessary
+func NewUTXOMapperSealed(key sdk.StoreKey, proto UTXOHolder) sealedUTXOMapper {
```

Comment:
Does anyone know why the SDK has a comment that the exposed codec is "not secure"?